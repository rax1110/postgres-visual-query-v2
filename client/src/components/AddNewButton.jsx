import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import React from 'react';
import * as PropTypes from 'prop-types';
import { Button } from 'reactstrap';

export const AddNewButton = ({ id, size, onClick }) => (
  <Button
    className=""
    outline
    color="info"
    size={size}
    id={id}
    onClick={onClick}
  >
    <FontAwesomeIcon icon="plus" />
  </Button>
);

AddNewButton.propTypes = {
  id: PropTypes.string,
  size: PropTypes.string,
  onClick: PropTypes.func,
};
