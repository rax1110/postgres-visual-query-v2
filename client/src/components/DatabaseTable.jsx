import React from 'react';
import { Button, Tooltip } from 'reactstrap';
import { connect } from 'react-redux';
import _ from 'lodash';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';
import { addColumn, addTable, removeTable } from '../actions/queryActions';
import { withToggle } from '../hocs/withToggle';

export const DatabaseTable = ({ data, checked, id, addTableProp, toggle, toggleStatus }) => {
  const handleOnClick = () => {
    addTableProp(data);
  };

  let tableTypeColor = 'primary';

  if (_.isEqual(data.table_type, 'VIEW')) {
    tableTypeColor = 'info';
  } else if (_.isEqual(data.table_type, 'FOREIGN')) {
    tableTypeColor = 'secondary';
  } else if (_.isEqual(data.table_type, 'MATERIALIZED VIEW')) {
    tableTypeColor = 'light';
  }

  const btnSelected = checked ? 'success' : tableTypeColor;

  const modifiers = {
    preventOverflow: {
      enabled: false,
    },
    hide: {
      enabled: false,
    },
  };

  return (
    <div className="w-100 pr-1">
      <Button
        size="sm"
        color={btnSelected}
        id={id}
        className="btn-block my-1 pt-0 text-left"
        onClick={handleOnClick}
      >
        <small
          color={tableTypeColor}
          className="text-truncate align-self-start"
        >
          <span className="mr-1 px-0">
            <FontAwesomeIcon icon="th-large" />
          </span>
          {data.table_type}
        </small>
        <div className="text-truncate">
          {` ${data.table_name}`}
        </div>
      </Button>
      <Tooltip
        placement="right"
        isOpen={toggleStatus}
        target={id}
        toggle={toggle}
        modifiers={modifiers}
        delay={{ show: 200, hide: 0 }}
        className=""
      >
        {data.table_name}
      </Tooltip>
    </div>
  );
};

DatabaseTable.propTypes = {
  data: PropTypes.shape({ table_type: PropTypes.string, table_name: PropTypes.string }),
  checked: PropTypes.bool,
  id: PropTypes.string,
  addTableProp: PropTypes.func,
  toggle: PropTypes.func,
  toggleStatus: PropTypes.bool,
};

const mapDispatchToProps = {
  addColumn,
  addTableProp: data => addTable(data),
  removeTable,
};

export default withToggle(connect(null, mapDispatchToProps)(DatabaseTable));
