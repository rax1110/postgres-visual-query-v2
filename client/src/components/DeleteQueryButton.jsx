import React from 'react';
import { Button } from 'reactstrap';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';
import { deleteQuery, setActiveQuery } from '../actions/queryActions';
import { deleteQueries, removeMainFromQueries } from '../actions/queriesActions';

export const DeleteQueryButton = (props) => {
  const handleOnClick = () => {
    props.deleteQuery();

    const mainQuery = props.queries.find(query => query.id === 0);

    if (mainQuery) {
      props.removeMainFromQueries();
      props.setActiveQuery(mainQuery);
    } else {
      props.deleteQueries();
    }
  };

  return (
    <Button color="danger" className="mr-2" onClick={handleOnClick}>
      <FontAwesomeIcon icon="trash" />
    </Button>
  );
};

DeleteQueryButton.propTypes = {
  deleteQuery: PropTypes.func,
  deleteQueries: PropTypes.func,
  queries: PropTypes.arrayOf(PropTypes.shape({})),
  removeMainFromQueries: PropTypes.func,
  setActiveQuery: PropTypes.func,
};

const mapStateToProps = store => ({
  queries: store.queries,
});

const mapDispatchToProps = {
  deleteQuery,
  deleteQueries,
  removeMainFromQueries,
  setActiveQuery,
};

export default connect(mapStateToProps, mapDispatchToProps)(DeleteQueryButton);
