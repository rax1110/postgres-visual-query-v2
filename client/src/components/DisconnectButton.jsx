import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import { Redirect } from 'react-router-dom';
import * as PropTypes from 'prop-types';
import { disconnect } from '../actions/hostActions';
import { translations } from '../utils/translations';

export const DisconnectButton = (props) => {
  const handleOnClick = () => {
    props.disconnect();
  };

  if (!props.connected) {
    return <Redirect to="/" />;
  }

  return (
    <Button size="lg" className="btn-block my-2 px-2" color="danger" onClick={handleOnClick}>
      {translations[props.language.code].sideBar.disconnectB}
    </Button>
  );
};

DisconnectButton.propTypes = {
  disconnect: PropTypes.func,
  connected: PropTypes.bool,
  language: PropTypes.shape({ code: PropTypes.string }),
};

const mapStateToProps = store => ({
  connected: store.host.connected,
  language: store.settings.language,
});

const mapDispatchToProps = {
  disconnect,
};

export default connect(mapStateToProps, mapDispatchToProps)(DisconnectButton);
