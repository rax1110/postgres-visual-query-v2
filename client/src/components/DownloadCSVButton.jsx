import React from 'react';
import { connect } from 'react-redux';
import { CSVLink } from 'react-csv';
import _ from 'lodash';
import { Button } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';

export const DownloadCSVButton = (props) => {
  const getHeaders = (result) => {
    if (_.isNull(result)) {
      return [];
    }

    const headers = [];

    result.fields.forEach((field) => {
      const header = {
        label: field.name,
        key: field.name,
      };

      headers.push(header);
    });

    return headers;
  };

  const getData = (result) => {
    if (_.isNull(result)) {
      return [];
    }
    return result.rows;
  };

  const disabled = _.isNull(props.result);

  return (
    <CSVLink
      className="mr-2"
      data={getData(props.result)}
      headers={getHeaders(props.result)}
      filename="result.csv"
    >
      <Button disabled={disabled}>
        <FontAwesomeIcon icon="download" />
        <div className="d-inline"> CSV</div>
      </Button>
    </CSVLink>
  );
};

DownloadCSVButton.propTypes = {
  result: PropTypes.shape({}),
};

const mapStateToProps = store => ({
  result: store.query.result,
});

export default connect(mapStateToProps)(DownloadCSVButton);
