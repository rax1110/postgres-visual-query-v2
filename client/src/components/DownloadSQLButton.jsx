import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';

export const DownloadSQLButton = (props) => {
  const downloadContent = (name, content) => {
    if (navigator.msSaveBlob) {
      const blobObject = new Blob([content], { type: 'text/plain' });

      window.navigator.msSaveOrOpenBlob(blobObject, name);
    } else {
      const aTag = document.createElement('a');
      const file = new Blob([content], { type: 'text/plain' });

      aTag.href = URL.createObjectURL(file);
      aTag.download = name;

      document.body.appendChild(aTag);
      aTag.click();
    }
  };

  return (
    <Button
      className="mr-2"
      onClick={() => downloadContent('select_query.sql', props.sql)}
    >
      <FontAwesomeIcon icon="download" />
      <div className="d-inline"> SQL</div>
    </Button>
  );
};

DownloadSQLButton.propTypes = {
  sql: PropTypes.string,
};

const mapStateToProps = store => ({
  sql: store.query.sql,
});

export default connect(mapStateToProps)(DownloadSQLButton);
