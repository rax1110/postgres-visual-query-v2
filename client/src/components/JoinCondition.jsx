import React from 'react';
import { Button, CustomInput, InputGroup, InputGroupAddon, InputGroupText, Row } from 'reactstrap';
import _ from 'lodash';
import { connect } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import * as PropTypes from 'prop-types';
import { updateJoin } from '../actions/queryActions';
import { translations } from '../utils/translations';

export const JoinCondition = (props) => {
  const handleMainColumnChange = (e) => {
    e.preventDefault();

    const value = JSON.parse(e.target.value);

    let condition = _.cloneDeep(props.condition);

    condition = {
      ...condition,
      main_column: value.column_name,
    };

    const conditions = _.cloneDeep(props.join.conditions);
    const conditionIndex = conditions.findIndex(_condition => _condition.id === value.id);

    conditions[conditionIndex] = condition;

    let join = _.cloneDeep(props.join);

    join = {
      ...join,
      conditions,
    };

    props.updateJoin(join);
  };

  const handleSecondaryTableChange = (e) => {
    e.preventDefault();

    const value = JSON.parse(e.target.value);

    let condition = _.cloneDeep(props.condition);

    condition = {
      ...condition,
      secondary_table: value.table,
      secondary_column: '',
    };

    const conditions = _.cloneDeep(props.join.conditions);
    const conditionIndex = conditions.findIndex(_condition => _condition.id === value.id);

    conditions[conditionIndex] = condition;

    let join = _.cloneDeep(props.join);

    join = {
      ...join,
      conditions,
    };

    props.updateJoin(join);
  };

  const handleSecondaryColumnChange = (e) => {
    e.preventDefault();

    const value = JSON.parse(e.target.value);

    let condition = _.cloneDeep(props.condition);

    condition = {
      ...condition,
      secondary_column: value.column_name,
    };

    const conditions = _.cloneDeep(props.join.conditions);
    const conditionIndex = conditions.findIndex(_condition => _condition.id === value.id);

    conditions[conditionIndex] = condition;

    let join = _.cloneDeep(props.join);

    join = {
      ...join,
      conditions,
    };

    props.updateJoin(join);
  };

  const handleRemove = () => {
    let conditions = _.cloneDeep(props.join.conditions);

    conditions = conditions.filter(condition => condition.id !== props.condition.id);

    let join = _.cloneDeep(props.join);

    join = {
      ...join,
      conditions,
    };

    props.updateJoin(join);
  };

  const defaultValue = {
    id: props.condition.id,
    table: {
      table_schema: '',
      table_name: '',
      table_alias: '',
    },
  };

  return (
    <Row form className="my-2">
      <div className="col-auto">
        <InputGroup size="sm">
          <InputGroupAddon addonType="prepend">
            <InputGroupText>{props.join.main_table.table_name}</InputGroupText>
          </InputGroupAddon>
          <CustomInput
            type="select"
            id="main_table_columns"
            onChange={handleMainColumnChange}
            defaultValue=""
          >
            <option
              key={`${props.condition.id}-main-column-null`}
              value=""
            >
              {translations[props.language.code].queryBuilder.joinConditionMainColumn}
            </option>
            {props.join.main_table.columns.map((column) => {
              const value = {
                id: props.condition.id,
                column_name: column.column_name,
              };

              return (
                <option
                  key={`${props.condition.id}-main-column-${column.column_name}`}
                  value={JSON.stringify(value)}
                >
                  {column.column_name}
                </option>
              );
            })}
          </CustomInput>
        </InputGroup>
      </div>
      <div className="col-auto align-self-center">
        <FontAwesomeIcon icon="equals" size="xs" />
      </div>
      <div className="col-auto">
        <InputGroup size="sm">
          <CustomInput
            bsSize="sm"
            type="select"
            id="secondary_table"
            className="text-secondary"
            onChange={handleSecondaryTableChange}
            defaultValue={JSON.stringify(defaultValue)}
          >
            <option
              key={`${props.condition.id}-secondary-table-null`}
              value={JSON.stringify(defaultValue)}
            >
              {translations[props.language.code].queryBuilder.joinConditionSecondaryTable}
            </option>
            {props.tables.map((table) => {
              const value = {
                id: props.condition.id,
                table,
              };
              const option = table.table_alias.length > 0 ? `${table.table_name} (${table.table_alias})` : `${table.table_name}`;

              return props.join.main_table.id !== table.id && (
                <option
                  key={`${props.condition.id}-secondary-table-${table.id}`}
                  value={JSON.stringify(value)}
                >
                  {option}
                </option>
              );
            })}
          </CustomInput>
          <CustomInput
            bsSize="sm"
            type="select"
            id="secondary_table_columns"
            className="text-secondary"
            onChange={handleSecondaryColumnChange}
            defaultValue=""
          >
            <option key={`${props.condition.id}-secondary-column-null`} value="">
              {translations[props.language.code].queryBuilder.joinConditionSecondaryColumn}
            </option>
            {!_.isEmpty(props.condition.secondary_table.table_name)
            && props.condition.secondary_table.columns.map((column) => {
              const value = {
                id: props.condition.id,
                column_name: column.column_name,
              };

              return (
                <option
                  key={`${props.condition.id}-secondary-column-${column.column_name}`}
                  value={JSON.stringify(value)}
                >
                  {column.column_name}
                </option>
              );
            })}
          </CustomInput>
        </InputGroup>
      </div>
      <div className="col-auto ml-auto">
        <Button size="sm" color="danger" onClick={handleRemove}>
          <FontAwesomeIcon icon="times" />
        </Button>
      </div>
    </Row>
  );
};

JoinCondition.propTypes = {
  id: PropTypes.string,
  language: PropTypes.shape({ code: PropTypes.string }),
  join: PropTypes.shape({
    id: PropTypes.number,
    conditions: PropTypes.array,
    main_table: PropTypes.shape({
      table_name: PropTypes.string,
      id: PropTypes.number,
      columns: PropTypes.arrayOf(PropTypes.shape({})),
    }),
    color: PropTypes.string,
    join_type: PropTypes.string,
  }),
  updateJoin: PropTypes.func,
  condition: PropTypes.shape({
    id: PropTypes.number,
    secondary_table: PropTypes.shape({
      table_name: PropTypes.string,
      columns: PropTypes.arrayOf(PropTypes.shape({})),
    }),
  }),
  tables: PropTypes.arrayOf(PropTypes.shape({
    table_alias: PropTypes.string,
    table_name: PropTypes.string,
    table_schema: PropTypes.string,
  })),
};

const mapStateToProps = store => ({
  tables: store.query.tables,
  language: store.settings.language,
});

const mapDispatchToProps = {
  updateJoin,
};

export default connect(mapStateToProps, mapDispatchToProps)(JoinCondition);
