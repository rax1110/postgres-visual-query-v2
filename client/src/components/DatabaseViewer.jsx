import React from 'react';
import { connect } from 'react-redux';
import { Scrollbars } from 'react-custom-scrollbars';
import _ from 'lodash';
import * as PropTypes from 'prop-types';
import DatabaseTable from './DatabaseTable';
import { filterTable } from '../utils/filterTable';

export const DatabaseViewer = (props) => {
  const constructData = (table) => {
    const data = {
      table_schema: table.table_schema,
      table_name: table.table_name,
      table_type: table.table_type,
      table_alias: '',
    };

    let constraints = JSON.parse(JSON.stringify(props.constraints));

    constraints = constraints.filter(constraint => constraint.table_schema === data.table_schema
      && constraint.table_name === data.table_name);

    let columns = JSON.parse(JSON.stringify(props.columns));

    columns = columns.filter(column => column.table_name === data.table_name
      && column.table_schema === data.table_schema).map((column) => {
      const col = column;

      col.constraints = constraints.filter(
        constraint => _.includes(constraint.column_name, column.column_name),
      );

      delete col.table_name;
      delete col.table_schema;
      return col;
    });

    data.columns = columns;

    return data;
  };

  return (
    <div className="flex-fill">
      <Scrollbars className="d-flex" autoHide>
        <div className="mt-1 pr-2">
          {props.tables.map((table, index) => {
            const checked = props.queryTable.some(
              queryTable => _.isEqual(table.table_name, queryTable.table_name)
                && _.isEqual(table.table_schema, queryTable.table_schema),
            );

            const id = `database-table-${index}`;

            return table.table_schema === props.selectedSchema
              && filterTable(table, props.searchExpr)
              && (
                <DatabaseTable
                  data={constructData(table)}
                  checked={checked}
                  key={id}
                  id={id}
                />
              );
          })}
        </div>
      </Scrollbars>
    </div>
  );
};

DatabaseViewer.propTypes = {
  data: PropTypes.shape({ table_type: PropTypes.string, table_name: PropTypes.string }),
  constraints: PropTypes.arrayOf(PropTypes.shape({})),
  columns: PropTypes.arrayOf(PropTypes.shape({})),
  tables: PropTypes.arrayOf(PropTypes.shape({})),
  queryTable: PropTypes.arrayOf(PropTypes.shape({})),
  selectedSchema: PropTypes.string,
  searchExpr: PropTypes.string,
};

const mapStateToProps = store => ({
  tables: store.database.tables,
  schemas: store.database.schemas,
  selectedSchema: store.database.selectedSchema,
  constraints: store.database.constraints,
  columns: store.database.columns,
  queryTable: store.query.tables,
  searchExpr: store.database.searchExpr,
});

export default connect(mapStateToProps)(DatabaseViewer);
