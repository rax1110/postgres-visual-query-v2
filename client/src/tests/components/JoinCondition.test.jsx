import React from 'react';
import { shallow } from 'enzyme';
import { JoinCondition } from '../../components/JoinCondition';

describe('Component: JoinCondition', () => {
  let component;
  let props;

  beforeEach(() => {
    props = {
      tables: [
        {
          id: 1,
          table_alias: '',
          table_name: 'amet',
          table_schema: 'public',
          table_type: 'BASE TABLE',
          columns: [
            {
              column_name: 'amet_kood',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 1,
              table_alias: '',
              table_id: 1,
              table_name: 'amet',
              table_schema: 'public',
            },
            {
              column_name: 'nimetus',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 2,
              table_alias: '',
              table_id: 1,
              table_name: 'amet',
              table_schema: 'public',
            },
            {
              column_name: 'kirjeldus',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 3,
              table_alias: '',
              table_id: 1,
              table_name: 'amet',
              table_schema: 'public',
            },
          ],
        },
        {
          id: 2,
          table_alias: '',
          table_name: 'asukoht',
          table_schema: 'public',
          table_type: 'BASE TABLE',
          columns: [
            {
              column_name: 'asukoht_kood',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 1,
              table_alias: '',
              table_id: 1,
              table_name: 'asukoht',
              table_schema: 'public',
            },
            {
              column_name: 'nimetus',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 2,
              table_alias: '',
              table_id: 1,
              table_name: 'asukoht',
              table_schema: 'public',
            },
          ],
        },
      ],
      join: {
        color: '#ddbf4f',
        conditions: [{
          id: 0,
          main_column: '',
          secondary_column: '',
          secondary_table: {
            table_alias: '',
            table_name: '',
            table_schema: '',
          },
        }],
        id: 0,
        main_table: {
          table_alias: '',
          table_name: '',
          table_schema: '',
          columns: [
            {
              column_name: 'asukoht_kood',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 1,
              table_alias: '',
              table_id: 2,
              table_name: 'asukoht',
              table_schema: 'public',
            },
            {
              column_name: 'nimetus',
              constraints: [],
              data_type: 'smallint',
              ordinal_position: 2,
              table_alias: '',
              table_id: 2,
              table_name: 'asukoht',
              table_schema: 'public',
            },
          ],
        },
        type: 'inner',
      },
      condition: {
        id: 0,
        main_column: '',
        secondary_column: '',
        secondary_table: {
          table_alias: '',
          table_name: '',
          table_schema: '',
        },
      },
      key: 'join-1-condition-1',
      language: { code: 'eng' },
      updateJoin: jest.fn(),
    };
  });

  test('JoinCondition renders with default props', () => {
    component = shallow(<JoinCondition {...props} />);

    expect(component).toMatchSnapshot();
  });

  test('JoinCondition handleMainColumnChange calls updateJoin once', () => {
    component = shallow(<JoinCondition {...props} />);

    const e = {
      target: {
        value: JSON.stringify({ id: 'id', column_name: 'asukoht' }),
        id: 1,
      },
      preventDefault: jest.fn(),
    };

    component.find('CustomInput').first().simulate('change', e);

    expect(props.updateJoin).toBeCalled();
  });

  test('JoinCondition handleSecondaryColumnChange calls updateJoin once', () => {
    component = shallow(<JoinCondition {...props} />);

    const e = {
      target: {
        value: JSON.stringify({ id: 'id', column_name: 'asukoht' }),
        id: 1,
      },
      preventDefault: jest.fn(),
    };

    component.find('CustomInput').at(2).simulate('change', e);

    expect(props.updateJoin).toBeCalled();
  });

  test('JoinCondition handleRemove calls updateJoin once', () => {
    component = shallow(<JoinCondition {...props} />);

    component.find('Button').simulate('click');

    expect(props.updateJoin).toBeCalled();
  });

  test('JoinCondition handleSecondaryTableChange calls updateJoin once', () => {
    component = shallow(<JoinCondition {...props} />);

    const e = {
      target: {
        value: JSON.stringify({
          id: 'id',
          table: {
            id: 1,
            table_alias: '',
            table_name: 'amet',
            table_schema: 'public',
            table_type: 'BASE TABLE',
            columns: [
              {
                column_name: 'amet_kood',
                constraints: [],
                data_type: 'smallint',
                ordinal_position: 1,
                table_alias: '',
                table_id: 1,
                table_name: 'amet',
                table_schema: 'public',
              },
              {
                column_name: 'nimetus',
                constraints: [],
                data_type: 'smallint',
                ordinal_position: 2,
                table_alias: '',
                table_id: 1,
                table_name: 'amet',
                table_schema: 'public',
              },
              {
                column_name: 'kirjeldus',
                constraints: [],
                data_type: 'smallint',
                ordinal_position: 3,
                table_alias: '',
                table_id: 1,
                table_name: 'amet',
                table_schema: 'public',
              },
            ],
          },
        }),
        id: 1,
      },
      preventDefault: jest.fn(),
    };

    component.find('CustomInput').at(1).simulate('change', e);

    expect(props.updateJoin).toBeCalled();
  });
});
