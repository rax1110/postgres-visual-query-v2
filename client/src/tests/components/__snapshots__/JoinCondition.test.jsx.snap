// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: JoinCondition JoinCondition renders with default props 1`] = `ShallowWrapper {}`;
